soh/src/code/z_scene_table.c

```diff
@@ -88,23 +89,23 @@ Gfx sDefaultDisplayList[] = {
 
 // Computes next entrance index based on age and day time to set the fade out transition
 void Scene_SetTransitionForNextEntrance(PlayState* play) {
-    s16 entranceIndex;
+    s16 layer = 0;
```

Comment:
With dynamic scenes comes dynamic entrances. We can no longer assume entrances are continuous, so this math needed to be reworked. This is the point of the EntranceDB_RetrieveLayer function.